syntax = "proto3";

option cc_generic_services = true;

//定义服务接口
service SdService {
    rpc text2img (SdText2ImgRequest) returns (SdResponse) {};
    rpc img2img (SdImg2ImgRequest) returns (SdResponse) {};
    rpc upscale (SdUpscaleRequest) returns (SdResponse) {};
    rpc imgfuse (SdImgFuseRequest) returns (SdResponse) {};

    rpc text2img_asyn (SdText2ImgRequest) returns (SdAsynTaskResponse) {};
    rpc img2img_asyn (SdImg2ImgRequest) returns (SdAsynTaskResponse) {};
    rpc upscale_asyn (SdUpscaleRequest) returns (SdAsynTaskResponse) {};
    rpc imgfuse_asyn (SdImgFuseRequest) returns (SdAsynTaskResponse) {};

    rpc query(SdQueryRequest) returns (SdResponse) {};
}
service SdEngine {
    rpc text2img (SdText2ImgRequest) returns (SdResponse) {};
    rpc img2img (SdImg2ImgRequest) returns (SdResponse) {};
    rpc upscale (SdUpscaleRequest) returns (SdResponse) {};
    rpc imgfuse (SdImgFuseRequest) returns (SdResponse) {}
}
//请求的参数
message SdText2ImgRequest {
    string prompt = 1;
    string negative_prompt = 2;
    int32 width = 3;
    int32 height= 4;
    int32 seed = 5;
    int32 steps = 6;
    int32 batch_size = 7;
    bool enable_hr = 8;
    float hr_scale = 9;
    string hr_upscaler = 10;
};

message SdImg2ImgRequest {
    repeated string base64_images = 1;
    string mask = 2;
    string prompt = 3;
    string negative_prompt = 4;
    int32 width = 5;
    int32 height= 6;
    int32 seed = 7;
    int32 steps = 8;
    int32 batch_size = 9;
    float denoising_strength = 10;
};

message SdUpscaleRequest {
    string base64_image = 1;
    int32 upscaling_resize = 2;
    string upscaler_1= 3;
};

message SdImgFuseRequest {
    repeated string base64_images = 1;
    string prompt = 2;
    string negative_prompt = 3;
    int32 width = 4;
    int32 height= 5;
    int32 seed = 6;
    int32 steps = 7;
    int32 batch_size = 8;
    float denoising_strength = 9;
};

//返回的对象
message SdResponse {
    string message = 1;
    int32 status = 2;
    repeated string base64 = 3;
}

message SdAsynTaskResponse {
    string message = 1;
    int32 status = 2;
    string task_id = 3;
}

message SdQueryRequest {
    string task_id = 1;
};
